ts/containers/staking/wizard/wizard.ts

```diff
@@ -28,10 +37,26 @@ const mapDispatchToProps = (dispatch: Dispatch<Action>): ConnectedDispatch => {
         onOpenConnectWalletDialog: (): void => {
             dispatcher.updateIsConnectWalletDialogOpen(true);
         },
+        onDepositAndStartStakingAsync: async (
+            providerState: ProviderState,
+            networkId: Network,
+            amountToStakeInput: string,
+        ): Promise<void> => {
+            const amountToStakeBaseUnits = new BigNumber(amountToStakeInput, 10).multipliedBy(constants.ZRX_BASE_UNIT);
+
+            const account = providerState.account as AccountReady;
+            const currentAllowance = account.zrxAllowanceBaseUnitAmount || new BigNumber(0);
+
+            if (amountToStakeBaseUnits.isGreaterThan(currentAllowance)) {
+                await asyncDispatcher.increaseZrxAllowanceAndDispatchToStoreIfNeededAsync(
+                    providerState,
+                    networkId,
+                    amountToStakeBaseUnits,
+                    dispatcher,
+                );
+            }
```

Comment:
In order to get the staking contract interactions working it seems like I need to upgrade the `0x` dependencies to the latest `beta` versions and this leads to a lot of errors due to api and type changes in packages. 

I will proceed with the upgrade in a separate branch as this will probably be quite significant changes just to get everything up and running again with the latest modules. After that I can circle back to actually hooking things up with the contract.